### PR TITLE
ci: split build job

### PR DESCRIPTION
**Pull request by emersion**
Split CI build job into multiple jobs. This should decrease build time (by using more workers in parallel for intensive tasks) and has the nice side-effect of making the build failure logs more readable (less noise to mentally filter out).

See https://github.com/OpenRailAssociation/osrd/issues/7559

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Looks good to me ! Thank you for doing it

**Review by woshilapin — APPROVED** [review]

**Review by Khoyo — APPROVED** [review]

**Comment by emersion** [comment]
It seems like the "build" job is marked as required for merging in the GitHub settings, and blocks this PR. I think it could be removed because other build steps that depend on it are required anyways (all of the tests). Alternatively, one could replace "build" with every job of the matrix.

**Comment by ghost** [comment]
> It seems like the "build" job is marked as required for merging in the GitHub settings, and blocks this PR. I think it could be removed because other build steps that depend on it are required anyways (all of the tests). Alternatively, one could replace "build" with every job of the matrix.

I think the rationale about removing the requirement makes sense. I can do it, I'll check with other DevOps before doing so and after I'll merge this PR.

**Comment by Khoyo** [comment]
Done

**Comment by emersion** [comment]
Thanks!

**Pull request merged**
